src/aten/sycl/Dropout.cpp

```diff
@@ -0,0 +1,495 @@
+                  mask_t>(
+                  self_info, ret_info, mask_info, nelem, pa, rng_engine_inputs);
+              sycl_kernel_submit(
+                  num_groups * group_size,
+                  group_size,
+                  getCurrentSYCLQueue(),
+                  caller);
+            } break;
+          }
+        } else {
+          switch (self_info.dims) {
+            case 1: {
+              auto caller = FusedDropoutUnrollFunctor<
+                  scalar_t,
+                  accscalar_t,
+                  index_type,
+                  1,
+                  1,
+                  mask_t>(
+                  self_info, ret_info, mask_info, nelem, pa, rng_engine_inputs);
+              sycl_kernel_submit(
+                  num_groups * group_size,
+                  group_size,
+                  getCurrentSYCLQueue(),
+                  caller);
+            } break;
+            default:
+              if (!self.is_contiguous() && ret.is_contiguous() &&
+                  mask.is_contiguous()) {
+                auto caller = FusedDropoutUnrollFunctor<
+                    scalar_t,
+                    accscalar_t,
+                    index_type,
+                    -1,
+                    1,
+                    mask_t>(
+                    self_info,
+                    ret_info,
+                    mask_info,
+                    nelem,
+                    pa,
+                    rng_engine_inputs);
+                sycl_kernel_submit(
+                    num_groups * group_size,
+                    group_size,
+                    getCurrentSYCLQueue(),
+                    caller);
+              } else {
+                auto caller = FusedDropoutUnrollFunctor<
+                    scalar_t,
+                    accscalar_t,
+                    index_type,
+                    -1,
+                    -1,
+                    mask_t>(
+                    self_info,
+                    ret_info,
+                    mask_info,
+                    nelem,
+                    pa,
+                    rng_engine_inputs);
+                sycl_kernel_submit(
+                    num_groups * group_size,
+                    group_size,
+                    getCurrentSYCLQueue(),
+                    caller);
+              }
+          }
+        }
+      });
+}
+
+} // namespace
+
+template <typename mask_t>
+std::tuple<Tensor, Tensor> dropout(
+    XPUGeneratorImpl* gen,
+    const Tensor& self,
+    double p) {
+  Tensor mask = at::empty_like(
+      self, self.options().dtype(c10::CppTypeToScalarType<mask_t>::value));
+  const int64_t nelem = self.numel();
+  // empty tensors should not get here, but just in case, avoid FPE
+  // non-training shot-cut
+  if (nelem == 0)
+    return std::tuple<Tensor, Tensor>(self.clone(), mask);
+
+  Tensor ret = at::empty_like(self);
+
+  auto group_size = std::min(
+      syclGpuHWThreadsPerEU() * syclMaxSubGroupSize(), syclMaxWorkGroupSize());
```

Comment:
Please give definite comments if we have to use `syclGpuHWThreadsPerEU() * syclMaxSubGroupSize()`. Since in people mind, when configuring SYCL work group size, people only regard concurrency.